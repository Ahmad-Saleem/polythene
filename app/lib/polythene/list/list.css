.list {
  padding: 8px 0; }
  .list.has-subheader {
    padding-top: 0; }
