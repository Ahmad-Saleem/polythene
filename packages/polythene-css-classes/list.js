import listTileClasses from "./list-tile";

export default {
  component:       "pe-list",

  // states
  borders:         "pe-list--borders",
  compact:         "pe-list--compact",
  hasHeader:       "pe-list--header",
  indentedBorders: "pe-list--indented-borders",
  padding:         "pe-list--padding",

  // lookup
  header:          listTileClasses.header
};
