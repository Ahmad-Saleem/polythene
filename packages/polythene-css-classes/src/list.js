import { listTileClasses } from "./list-tile";

export const listClasses = {
  component:       "pe-list",

  // elements
  header:          listTileClasses.header,

  // states
  borders:         "pe-list--borders",
  compact:         "pe-list--compact",
  hasHeader:       "pe-list--header",
  indentedBorders: "pe-list--indented-borders",
  padding:         "pe-list--padding",
};
