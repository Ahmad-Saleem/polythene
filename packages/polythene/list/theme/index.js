import { styler } from 'polythene-core-css';
import { componentConfig } from 'polythene-theme';
import vars from './config';
import layout from './layout';
import color from './color';

var configFn = componentConfig && componentConfig.list;
var config = configFn ? configFn(vars) : vars;
var id = 'pe-list';

styler.add(id, layout(config), color(config));