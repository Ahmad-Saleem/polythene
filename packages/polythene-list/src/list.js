import m from "mithril";
import listTile from "polythene-list-tile";
import { filterSupportedAttributes } from "polythene-core";
import { customTheme } from "./theme";

export const classes = {
  component:       "pe-list",
  header:          "pe-list__header",
  borders:         "pe-list--borders",
  indentedBorders: "pe-list--indented-borders",
  hasHeader:       "pe-list--header",
  compact:         "pe-list--compact"
};

const view = vnode => {
  const attrs = vnode.attrs;
  const element = attrs.element || "div";
  const props = Object.assign(
    {},
    filterSupportedAttributes(attrs),
    {
      class: [
        classes.component,
        attrs.borders ? classes.borders : null,
        attrs.indentedBorders ? classes.indentedBorders : null,
        attrs.header ? classes.hasHeader : null,
        attrs.compact ? classes.compact : null,
        attrs.tone === "dark" ? "pe-dark-theme" : null,
        attrs.tone === "light" ? "pe-light-theme" : null,
        attrs.class
      ].join(" "),
    }
  );
  let headerOpts;
  if (attrs.header) {
    headerOpts = Object.assign({}, attrs.header);
    headerOpts.class = [
      classes.header,
      headerOpts.class || null
    ].join(" ");
  }
  const content = [
    headerOpts ? m(listTile, headerOpts) : null,
    attrs.tiles
      ? attrs.tiles
      : attrs.content
        ? attrs.content
        : attrs.children || vnode.children
  ];
  return m(element, props, [attrs.before, content, attrs.after]);
};

export default {
  theme: customTheme, // accepts (selector, vars)
  view
};
