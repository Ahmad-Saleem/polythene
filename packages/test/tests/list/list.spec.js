/* global describe, it, expect */
import m from "mithril";
import { tidy, defaultHtmlTidyOptions } from "mithril-jest";
import { tests } from "./tests.js";
import { runSnapshots } from "../../scripts/snapshots";
import list from "polythene-list";

runSnapshots(tests);

describe("List component", () => {
  it("no options", () => {
    const cmp = m(list);
    const html = tidy(cmp, {
      ...defaultHtmlTidyOptions,
      wrap: false,
      indent: false
    });
    expect(html).toContain("<div class=\"pe-list     \"></div>");
  });
  it("child", () => {
    const cmp = m(list, {}, m("span", "Child"));
    const html = tidy(cmp);
    expect(html).toContain("<span>Child</span>");
  });
  it("option header", () => {
    const cmp = m(list, {header: m("span", "Header")});
    const html = tidy(cmp);
    expect(html).toContain("<span>Header</span>");
  });
  it("option id", () => {
    const cmp = m(list, {id: "id-x"});
    const html = tidy(cmp);
    expect(html).toContain("id=\"id-x\"");
  });
  it("option class", () => {
    const cmp = m(list, {class: "class-x"});
    const html = tidy(cmp);
    expect(html).toContain("class-x\">");
  });
  it("option element", () => {
    const cmp = m(list, {element: "span"});
    const html = tidy(cmp);
    expect(html).toContain("<span ");
    expect(html).toContain("</span>");
  });
  it("option before", () => {
    const cmp = m(list, {before: m("span", "Before")});
    const html = tidy(cmp);
    expect(html).toContain("<span>Before</span>");
  });
  it("option after", () => {
    const cmp = m(list, {after: m("span", "After")});
    const html = tidy(cmp);
    expect(html).toContain("<span>After</span>");
  });
});
