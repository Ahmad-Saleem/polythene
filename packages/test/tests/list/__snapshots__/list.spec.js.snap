exports[`test Child nodes 1`] = `
"<div class=\"pe-list pe-list--borders\">
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`test No options 1`] = `
"<div class=\"pe-list\"></div>
"
`;

exports[`test Option: after 1`] = `
"<div class=\"pe-list pe-list--header\">
  <div class=\"pe-list-tile pe-list__header\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          My header
        </div>
      </div>
    </div>
  </div>
  <div>
    After
  </div>
</div>
"
`;

exports[`test Option: before 1`] = `
"<div class=\"pe-list pe-list--header\">
  <div>
    Before
  </div>
  <div class=\"pe-list-tile pe-list__header\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          My header
        </div>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`test Option: class 1`] = `
"<div class=\"pe-list class-x\"></div>
"
`;

exports[`test Option: compact 1`] = `
"<div class=
\"pe-list pe-list--borders pe-list--header pe-list--compact\">
  <div class=\"pe-list-tile pe-list__header\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Friends
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <a href=\"#!/\" class=\"pe-list-tile__primary\">
    <div class=
    \"pe-list-tile__content pe-list-tile__content--front\">
      <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
        <img src=
        \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
      </div>
    </div>
    <div class=\"pe-list-tile__content\">
      <div class=\"pe-list-tile__title\">
        Jennifer Barker
      </div>
      <div class=\"pe-list-tile__subtitle\">
        Starting post doc
      </div>
    </div></a>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`test Option: element 1`] = `
"<blockquote class=\"pe-list\"></blockquote>
"
`;

exports[`test Option: header 1`] = `
"<div class=\"pe-list pe-list--header\">
  <div class=\"pe-list-tile pe-list__header\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          My header
        </div>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`test Option: hoverable -- dark theme 1`] = `
"<div class=
\"pe-list pe-list--borders pe-list--hoverable pe-list--header\">
  <div class=\"pe-list-tile pe-list__header\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Friends
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`test Option: hoverable 1`] = `
"<div class=
\"pe-list pe-list--borders pe-list--hoverable pe-list--header\">
  <div class=\"pe-list-tile pe-list__header\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Friends
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`test Option: id 1`] = `
"<div id=\"id-x\" class=\"pe-list\"></div>
"
`;

exports[`test Option: selectable 1`] = `
"<div class=
\"pe-list pe-list--borders pe-list--selectable pe-list--header\">
  <div class=\"pe-list-tile pe-list__header\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Friends
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`test Option: tiles 1`] = `
"<div class=\"pe-list pe-list--borders\">
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`test Options: header, tiles with urls 1`] = `
"<div class=\"pe-list pe-list--borders pe-list--header\">
  <div class=\"pe-list-tile pe-list__header\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Friends
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <a href=\"#!/\" class=\"pe-list-tile__primary\">
    <div class=
    \"pe-list-tile__content pe-list-tile__content--front\">
      <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
        <img src=
        \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
      </div>
    </div>
    <div class=\"pe-list-tile__content\">
      <div class=\"pe-list-tile__title\">
        Jennifer Barker
      </div>
      <div class=\"pe-list-tile__subtitle\">
        Starting post doc
      </div>
    </div></a>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <a href=\"#!/\" class=\"pe-list-tile__primary\">
    <div class=
    \"pe-list-tile__content pe-list-tile__content--front\">
      <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
        <img src=
        \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
      </div>
    </div>
    <div class=\"pe-list-tile__content\">
      <div class=\"pe-list-tile__title\">
        Ali Connors
      </div>
      <div class=\"pe-list-tile__subtitle\">
        Brunch this weekend?
      </div>
    </div></a>
  </div>
</div>
<div class=\"pe-list pe-list--borders pe-list--header\">
  <div class=\"pe-list-tile pe-list__header\">
    <div class=\"pe-list-tile__primary\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Friends
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <a href=\"#!/\" class=\"pe-list-tile__primary\">
    <div class=
    \"pe-list-tile__content pe-list-tile__content--front\">
      <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
        <img src=
        \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
      </div>
    </div>
    <div class=\"pe-list-tile__content\">
      <div class=\"pe-list-tile__title\">
        Jennifer Barker
      </div>
      <div class=\"pe-list-tile__subtitle\">
        Starting post doc
      </div>
    </div></a>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <a href=\"#!/\" class=\"pe-list-tile__primary\">
    <div class=
    \"pe-list-tile__content pe-list-tile__content--front\">
      <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
        <img src=
        \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
      </div>
    </div>
    <div class=\"pe-list-tile__content\">
      <div class=\"pe-list-tile__title\">
        Ali Connors
      </div>
      <div class=\"pe-list-tile__subtitle\">
        Brunch this weekend?
      </div>
    </div></a>
  </div>
</div>
"
`;

exports[`test Options: header.sticky 1`] = `
"<div class=\"scrollable-list\">
  <div class=\"pe-list pe-list--header\">
    <div class=\"pe-list-tile pe-list-tile--sticky pe-list__header\">
      <div class=\"pe-list-tile__primary\">
        <div class=\"pe-list-tile__content\">
          <div class=\"pe-list-tile__title\">
            Subheader
          </div>
        </div>
      </div>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div></a>
    </div>
  </div>
  <div class=\"pe-list pe-list--header\">
    <div class=\"pe-list-tile pe-list-tile--sticky pe-list__header\">
      <div class=\"pe-list-tile__primary\">
        <div class=\"pe-list-tile__content\">
          <div class=\"pe-list-tile__title\">
            Subheader
          </div>
        </div>
      </div>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div></a>
    </div>
  </div>
  <div class=\"pe-list pe-list--header\">
    <div class=\"pe-list-tile pe-list-tile--sticky pe-list__header\">
      <div class=\"pe-list-tile__primary\">
        <div class=\"pe-list-tile__content\">
          <div class=\"pe-list-tile__title\">
            Subheader
          </div>
        </div>
      </div>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div></a>
    </div>
  </div>
  <div class=\"pe-list pe-list--header\">
    <div class=\"pe-list-tile pe-list-tile--sticky pe-list__header\">
      <div class=\"pe-list-tile__primary\">
        <div class=\"pe-list-tile__content\">
          <div class=\"pe-list-tile__title\">
            Subheader
          </div>
        </div>
      </div>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div></a>
    </div>
  </div>
  <div class=\"pe-list pe-list--header\">
    <div class=\"pe-list-tile pe-list-tile--sticky pe-list__header\">
      <div class=\"pe-list-tile__primary\">
        <div class=\"pe-list-tile__content\">
          <div class=\"pe-list-tile__title\">
            Subheader
          </div>
        </div>
      </div>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-1.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div></a>
    </div>
    <div class=\"pe-list-tile pe-list-tile--subtitle\">
      <a href=\"#!/\" class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\">
        <div class=\"pe-icon pe-icon--large pe-icon--avatar\">
          <img src=
          \"http://arthurclemens.github.io/assets/polythene/examples/avatar-2.png\">
        </div>
      </div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div></a>
    </div>
  </div>
</div>
"
`;

exports[`test Options: tiles, indent, indentedBorders 1`] = `
"<div class=\"pe-list pe-list--indented-borders pe-list--header\">
  <div class=\"pe-list-tile pe-list-tile--front pe-list__header\">
    <div class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\"></div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Friends
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\"></div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\">
      <div class=
      \"pe-list-tile__content pe-list-tile__content--front\"></div>
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div>
    </div>
  </div>
</div>
"
`;
